Packs/jamf/Integrations/jamfV2/jamfV2.py

```diff
@@ -28,6 +29,73 @@
+            the client will use basic auth instead.
+        """
+        try:
+            self._token = self._get_token()
+            self._headers = {
+                'Authorization': f'Bearer {self._token}'
+            }
+            add_sensitive_log_strs(self._token)
+        except DemistoException as e:
+            demisto.log(str(e))
+            demisto.log("Couldn't create token will proceed using basic auth")
+
+    def _get_token(self) -> str:
+        """
+        Obtains token from integration context if available and still valid.
+        After expiration, new token are generated and stored in the integration context.
+        Returns:
+            str: token that will be added to authorization header.
+        """
+        integration_context = get_integration_context()
+        token = integration_context.get('token', '')
+        valid_until = integration_context.get('expires')
+
+        now_timestamp = arg_to_datetime('now').timestamp()  # type:ignore
+        # if there is a key and valid_until, and the current time is smaller than the valid until
+        # return the current token
+        if token and valid_until:
+            if now_timestamp < valid_until:
+                return token
+
+        # else generate a token and update the integration context accordingly
+        token = self._generate_token()
+
+        return token
+
+    def _generate_token(self) -> str:
+        """
+        Generates new token.
+        """
+        resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json')
+        self._token = resp.get('token')
+        expiration_time = int(dateparser.parse(resp.get('expires')).timestamp())
+        integration_context = get_integration_context()
+        integration_context.update({'token': self._token})
+        # Add 10 minutes buffer for the token
+        integration_context.update({'expires': expiration_time - 600})
+        set_integration_context(integration_context)
+
+        return self._token
+
+    def _classic_api_post(self, url_suffix, data, error_handler):
+        post_headers = ((self._headers or {}) | POST_HEADERS)  # merge the token and the POST headers
+        classic_url_suffix = urljoin('/JSSResource', url_suffix)  # classic API endpoints starts with JSSResource
+        return self._http_request(method='POST', data=data, url_suffix=classic_url_suffix,
+                                  headers=post_headers, resp_type='response',
+                                  error_handler=error_handler)
+
+    def _classic_api_get(self, url_suffix, error_handler=None):
+        get_headers = ((self._headers or {}) | GET_HEADERS)  # merge the token and the GET headers
+        classic_url_suffix = urljoin('/JSSResource', url_suffix)  # classic API endpoints starts with JSSResource
+        return self._http_request(method='GET', url_suffix=classic_url_suffix, headers=get_headers,
+                                  error_handler=error_handler)
```

Comment:
All of this is because of the token? I don't think we need it. You can just create a new function in your client `_http_request` that will override the original `_http_request` function in the base client. In your new function just add the token to the heders and call super().

Reply:
Jamf has two APIs, classic and pro. I think merging the two functions under ```_http_request``` would cause confusion to other developer if enhancement or maintenance is required.

Reply:
Not sure I am following, where do we use the new one, in the `_generate_token`? what is different there?

Reply:
I think the code is easier to maintain with those new methods, it is plausible that new commands (using the other api) would be added in the future. Using methods which explicitly define the API type could prevent some confusion.